cmd/pmm-managed-starlark/main.go

```diff
@@ -46,4 +58,59 @@ func main() {
 	if on, _ := strconv.ParseBool(os.Getenv("PMM_TRACE")); on {
 		logrus.SetLevel(logrus.TraceLevel)
 	}
+
+	l := logrus.WithField("component", "pmm-managed-starlark")
+
+	decoder := json.NewDecoder(os.Stdin)
+	var data checks.StarlarkScriptData
+	err := decoder.Decode(&data)
+	if err != nil {
+		l.Errorf("Error decoding json data: %s", err)
+		os.Exit(1)
+	}
+
+	err = runChecks(l, data)
+	if err != nil {
+		l.Errorf("Error running starlark script: %s", err)
+		os.Exit(1)
+	}
+}
+
+func runChecks(l *logrus.Entry, data checks.StarlarkScriptData) error {
+	err := unix.Setrlimit(unix.RLIMIT_CPU, &unix.Rlimit{Cur: cpuLimit, Max: cpuLimit})
+	if err != nil {
+		l.Warnf("Failed to limit CPU usage: %s", err)
+	}
+	err = unix.Setrlimit(unix.RLIMIT_DATA, &unix.Rlimit{Cur: memoryLimit, Max: memoryLimit})
```

Comment:
Are you sure it should be RLIMIT_DATA?

Reply:
i was trying to decide between `RLIMIT_DATA` and `RLIMIT_MEMLOCK`

From the manpage

```
RLIMIT_DATA
    The maximum size of the process's data segment (initialized data, uninitialized data, and heap).
```

> (initialized data, uninitialized data, and heap)

this :point_up: seemed to me like what you were referring to when talking about the process's memory

and 

I couldn't fully understand what `RLIMIT_MEMLOCK` was for from the manual
```
RLIMIT_MEMLOCK
    The maximum number of bytes of memory that may be locked into RAM. In effect this limit is rounded down to the nearest multiple of the system page size. This limit affects mlock(2) and mlockall(2) and the mmap(2) MAP_LOCKED operation. Since Linux 2.6.9 it also affects the shmctl(2) SHM_LOCK operation, where it sets a maximum on the total bytes in shared memory segments (see shmget(2)) that may be locked by the real user ID of the calling process. The shmctl(2) SHM_LOCK locks are accounted for separately from the per-process memory locks established by mlock(2), mlockall(2), and mmap(2) MAP_LOCKED; a process can lock bytes up to this limit in each of these two categories. In Linux kernels before 2.6.9, this limit controlled the amount of memory that could be locked by a privileged process. Since Linux 2.6.9, no limits are placed on the amount of memory that a privileged process may lock, and this limit instead governs the amount of memory that an unprivileged process may lock. 
```

do you want to set some other kind of limit for the memory? if so please point it out to me.

Reply:
Add me to the "Linux is a mess" camp – why on Earth data segment includes heap?! Everywhere else data, bss, and heap are in separate sections…

Ok, let's use RLIMIT_DATA